File: unused/TitleView.m
//
//  TitleView.m
//  Local
//
//  Created by Rostyslav.Stepanyak on 4/28/17.
//  Copyright © 2017 Tilf AB. All rights reserved.
//

#import "TitleView.h"
#import "Local-Swift.h"

#define DEGREES_TO_RADIANS(d) (d * M_PI / 180)

typedef NS_ENUM(NSInteger, DropDownState) {
    DropDownStateHidden = 0,
    DropDownStateRevealed = 1,
};

@interface TitleView()
@property (nonatomic, weak) IBOutlet UIButton *coverButton;
@property (nonatomic, weak) IBOutlet UILabel *mainLabel;
@property (nonatomic, weak) IBOutlet UILabel *dropDownLabel;
@property (nonatomic) DropDownState state;
@end

@implementation TitleView

- (IBAction)menuCLicked:(id)sender {
    [self.delegate titleViewClicked:self.state == DropDownStateHidden];
    
    if (self.state == DropDownStateHidden) {
        [self reveal];
    }
    
    if (self.state == DropDownStateRevealed) {
        [self collapse];
    }
}

- (void)reveal {
    self.coverButton.userInteractionEnabled = false;
    [UIView animateWithDuration:0.5 animations:^{
        self.dropDownLabel.transform = CGAffineTransformMakeRotation(DEGREES_TO_RADIANS(180));
    } completion:^(BOOL finished) {
        self.coverButton.userInteractionEnabled = true;
        self.state = DropDownStateRevealed;
    }];
}

- (void)collapse {
    self.coverButton.userInteractionEnabled = false;
    [UIView animateWithDuration:0.5 animations:^{
        self.dropDownLabel.transform = CGAffineTransformMakeRotation(DEGREES_TO_RADIANS(360));
    } completion:^(BOOL finished) {
        self.coverButton.userInteractionEnabled = true;
        self.state = DropDownStateHidden;
    }];
}

@end

File: local/Views/TitleView/TitleView.swift
//
//  TitleView.swift
//  Local
//
//  Created by Douglas Galante on 7/31/17.
//  Copyright © 2017 Tilf AB. All rights reserved.
//

import UIKit

enum DropDownState : Int {
    case hidden = 0
    case revealed = 1
}

class TitleView: UIView {
    weak var delegate: TitleViewDelegate?
    
    @IBOutlet weak var coverButton: UIButton!
    @IBOutlet weak var mainLabel: UILabel!
    @IBOutlet weak var dropDownLabel: UILabel!
    var state = DropDownState(rawValue: 0)!
    
    func reveal() {
        coverButton.isUserInteractionEnabled = false
        UIView.animate(withDuration: 0.5, animations: {() -> Void in
            self.dropDownLabel.transform = CGAffineTransform(rotationAngle: self.degreesToRadians(d: 180))
        }, completion: {(_ finished: Bool) -> Void in
            self.coverButton.isUserInteractionEnabled = true
            self.state = .revealed
        })
    }
    
    func collapse() {
        coverButton.isUserInteractionEnabled = false
        UIView.animate(withDuration: 0.5, animations: {() -> Void in
            self.dropDownLabel.transform = CGAffineTransform(rotationAngle: self.degreesToRadians(d: 360))
        }, completion: {(_ finished: Bool) -> Void in
            self.coverButton.isUserInteractionEnabled = true
            self.state = .hidden
        })
    }
    
    @IBAction func menuCLicked(_ sender: Any) {
        delegate?.titleViewClicked(state == .hidden)
        if state == .hidden {
            reveal()
        }
        if state == .revealed {
            collapse()
        }
    }
    
    func degreesToRadians(d: CGFloat) -> CGFloat {
        return d * .pi / 180
    }
}
